### ISSUE TITLE
Sequence diagram: Wrong start/end of activation boxes in activation boxes while using `teoz`

### ISSUE THREAD

**Issue by HeinrichAD** [plantuml/plantuml#1494]
**Describe the bug**
While using `teoz` in sequence diagrams, the activation boxes are shifted or do not start/end at the correct position.

| Current behavior |  Problems | Expected behavior |
:--:|:--:|:--:
| ![Current behavior](https://github.com/plantuml/plantuml/assets/5962361/54411408-52d9-4e95-b462-7b7c00695bdc) | ![Problems](https://github.com/plantuml/plantuml/assets/5962361/221bb927-1fe5-46c8-8048-871da8e80069) | ![Expected behavio](https://github.com/plantuml/plantuml/assets/5962361/fa778344-e3c4-4bda-9164-1ee49acb7a4c) |

**To Reproduce**
[Example code rendered on plantuml.com](https://www.plantuml.com/plantuml/uml/JOyz3eCm38Ltdy8iNH0Nw820gKiFuGIhYfeGv9XAzFJb9urOBBttlMVBNL8INOA7QXPw1KBb-4MLXI6ztJIJKC2ag-TjYhD75dghKV19taTPaHA6R0tHxD9asNjdUJS8WAow3oblq63JjaZOD7MDTtouCrW17S3ymR3olh62qYXsBjpm9E5QoGn6BZVfAW9qF8tR8py0)

```plantuml
@startuml
!pragma teoz true
skinparam style strictuml

Actor Mallory as m
Actor Bob as b
Actor Alice as a

activate b
b ->> a --++ : Hi Alice!
a ->> b --++ : Bye Bob!

b ->> a ++ : Bye Alice!
& b ->> m --++ : Bye Alice!

deactivate a
deactivate m
@enduml
```

**Desktop**

- OS: Windows 10
- Browser: Firefox
- Version: 102.12.0esr (64-Bit)

**Comment by jubr** [comment]
This teoz bug has been around for a while, see https://github.com/plantuml/plantuml/issues/1099. Could you please take a look? Tnx so much!

**Comment by jimnelson372** [comment]
Looking into these problems now.   I have a fix for the --++ condition, but I'm working on the & conditions.  So, for the moment, my fix improves it to:

![testteozlifeline_006](https://github.com/plantuml/plantuml/assets/66284321/1248bd78-d0c4-4499-ba56-5c103734a3ef)

So just that part on Bob after the Bye Alice messages is still a problem.   Again, working on it now.

**Comment by The-Lum** [comment]
Hi all, and @jimnelson372,

This will also fix:
- https://forum.plantuml.net/13409/teoz-arrow-alignment

Thanks for your works,
Regards,
Th.

**Comment by jimnelson372** [comment]
Yes, thanks The-Lum.  I had actually started with Forum issue 13409, but I didn't have 15191 in my test set.   Here's how they look with my changes:
13409:

![testteozlifeline_003](https://github.com/plantuml/plantuml/assets/66284321/e9b68c6e-1d3c-480f-b3fb-a11cb114cb2a)

15191:

![testteoz](https://github.com/plantuml/plantuml/assets/66284321/7e60e766-475d-4354-8e47-85e25259fd7d)

I have a partial fix for the current issue (the & condition not deactivating Bob's timeline), but I'm making it more robust.  Right now it only works for a pair of parallel messages, not for 3 or more.  I think I can get it all to work shortly.  Here's how the current issue diagram now looks:

![testteoz](https://github.com/plantuml/plantuml/assets/66284321/edeb8da9-a084-44ad-ad1e-462230f03881)

**Comment by jimnelson372** [comment]
Nice to have 15191, as it shows me I still have one more place (at least) to look for what had been a late starting activation on style 1.   You can see the remaining problem in that the style 1's end where the deactivation occurs while style 2 starts when the deactivation occurs.   I looking to make that consistent.

**Comment by The-Lum** [comment]
👀 See also:
- https://forum.plantuml.net/14803/inline-multicast-activate-only-activates-the-first-target

**Comment by The-Lum** [comment]
That will fix also:
- https://forum.plantuml.net/16663/small-rendering-difference-between-default-and-teoz

**Comment by jimnelson372** [comment]
Thanks @The-Lum.  I've been working on the problem reported in #656, and #1099, where depending on comparative number of rows of text in each message, the active areas can start early or end early.   I see where it is happening in the code, but I haven't yet worked out a solution.   Hopefully I can resolve several of related issues all at once, but if run into difficulties on some of them, I will go ahead and create a pull request for those I have fixed.

**Comment by jimnelson372** [comment]
Since I've been commenting about related fixes here, I just wanted to share that though I haven't yet solved issues #656 and #1099, I did find the fix for #739 while I was working on them.
